src/packagedcode/build.py

```diff
@@ -374,54 +375,27 @@ def parse(cls, location, package_only=True):
-            package_data = dict(
-                datasource_id=cls.datasource_id,
-                type=metadata_fields.get('upstream_type', cls.default_package_type),
-                name=metadata_fields.get('name'),
-                version=metadata_fields.get('version'),
-                extracted_license_statement=metadata_fields.get('licenses', []),
-                parties=parties,
-                homepage_url=metadata_fields.get('upstream_address', ''),
-                # TODO: Store 'upstream_hash` somewhere
-            )
-            yield models.PackageData.from_data(package_data, package_only=True)
-
-        if (
-            'package_type'
-            and 'name'
-            and 'version'
-            and 'license_expression'
-            and 'homepage_url'
-            and 'download_url'
-            and 'vcs_url'
-            and 'download_archive_sha1'
-            and 'vcs_commit_hash'
-            in metadata_fields
-        ):
-            package_data = dict(
-                datasource_id=cls.datasource_id,
-                type=metadata_fields.get('package_type', cls.default_package_type),
-                name=metadata_fields.get('name'),
-                version=metadata_fields.get('version'),
-                extracted_license_statement=metadata_fields.get('license_expression', ''),
-                parties=parties,
-                homepage_url=metadata_fields.get('homepage_url', ''),
-                download_url=metadata_fields.get('download_url', ''),
-                vcs_url=metadata_fields.get('vcs_url', ''),
-                sha1=metadata_fields.get('download_archive_sha1', ''),
-                extra_data=dict(vcs_commit_hash=metadata_fields.get('vcs_commit_hash', ''))
-            )
-            yield models.PackageData.from_data(package_data, package_only=True)
+        # TODO: Create function that determines package type from download URL,
+        # then create a package of that package type from the metadata info
+        package_data = dict(
+            datasource_id=cls.datasource_id,
+            type=metadata_fields.get('upstream_type', metadata_fields.get('package_type', cls.default_package_type)),
+            name=metadata_fields.get('name'),
+            version=metadata_fields.get('version'),
+            extracted_license_statement=metadata_fields.get('licenses', metadata_fields.get('license_expression')),
+            parties=parties,
+            homepage_url=metadata_fields.get('upstream_address', metadata_fields.get('homepage_url')),
+            download_url=metadata_fields.get('download_url'),
+            vcs_url=metadata_fields.get('vcs_url'),
+            sha1=metadata_fields.get('download_archive_sha1'),
+            # TODO: Store 'upstream_hash` somewhere
+        )
+        if 'vcs_commit_hash' in metadata_fields:
+            package_data["extra_data"] = dict(vcs_commit_hash=metadata_fields['vcs_commit_hash'])
```

Comment:
Wouldn't it be better to do a get() rather than checking if this exists in the mapping?

Reply:
When including the `upstream_hash`, I rewrote this part. Now we always create the `extra_data` dict with all keys and I use `.get` to extract the values.